ERCS/erc-7887.md

```diff
@@ -0,0 +1,378 @@
+---
+eip: 7887
+title: Cancelation for ERC-7540 Tokenized Vaults
+description: Extension of ERC-7540 with cancelation support
+author: Jeroen Offerijns (@hieronx), Vikram Arun (@vikramarun)
+discussions-to: https://ethereum-magicians.org/t/erc-7887-cancelation-for-erc-7540-tokenized-vaults/22906
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-02-18
+requires: 165, 7540
+---
+
+## Abstract
+
+The following standard extends [ERC-7540](./eip-7540.md) by adding support for asynchronous cancelation flows.
+
+## Motivation
+
+Shares or assets locked for Requests can be stuck in the Pending state. For some use cases, such as redeeming from a pool of long-dated real-world assets, this can take a considerable amount of time.
+
+This standard expands the scope Asynchronous ERC-7540 Vaults by adding cancelation support.
+
+## Specification
+
+### Definitions:
+
+The existing definitions from [ERC-7540](./eip-7540.md) apply.
+
+### Cancelation Lifecycle
+
+After submission, cancelation Requests go through Pending, Claimable, and Claimed stages. An example lifecycle for a deposit cancelation Request is visualized in the table below.
+
+| **State**   | **User**                         | **Vault** |
+|-------------|---------------------------------|-----------|
+| Pending     | `cancelDepositRequest(requestId, controller)` | `pendingCancelDepositRequest[controller] = true` |
+| Claimable   |                                 | *Internal cancelation fulfillment*:  `pendingCancelDepositRequest[controller] = false`; `claimableCancelDepositRequest[controller] = assets` |
+| Claimed     | `claimCancelDepositRequest(requestId, receiver, controller)`      | `claimableDepositRequest[controller] -= assets`; `asset.balanceOf[receiver] += assets` |
```

Comment:
The third column here looks like it depends on specific implementation details (`pendingCancelDepositRequest` looks like a `mapping`). Ideally everything in the Specification section should be independent of how the interface is implemented. That said, if there's no effective way to illustrate the flow without these details, they can stay, but you should explain where `pendingCancelDepositRequest` is defined.

Reply:
This table matches the table in the ERC-7540 spec. I think for the reader it would be helpful to be consistent, as an implementor of ERC-7887 would also need to implement ERC-7540.

I'll add a reference to where they are defined.